extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/VertxVaultClient.java

```diff
@@ -74,40 +74,52 @@
 import io.quarkus.vault.runtime.client.dto.transit.VaultTransitVerify;
 import io.quarkus.vault.runtime.client.dto.transit.VaultTransitVerifyBody;
 import io.quarkus.vault.runtime.config.VaultBootstrapConfig;
-import okhttp3.HttpUrl;
-import okhttp3.MediaType;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.RequestBody;
-import okhttp3.Response;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.ext.web.client.HttpRequest;
+import io.vertx.mutiny.ext.web.client.HttpResponse;
+import io.vertx.mutiny.ext.web.client.WebClient;
 
 @Singleton
-public class OkHttpVaultClient implements VaultClient {
+public class VertxVaultClient implements VaultClient {
```

Comment:
(In another PR)

We may be able to provide a reactive version (ReactiveVaultClient) which would produce `Uni`.
With this one, the imperative (current) version would only delegate and append `await().atMost(...)`.